src/fides/api/ops/service/masking/strategy/masking_strategy_hash.py

```diff
@@ -58,7 +59,11 @@ def mask(
 
         masked_values: List[str] = []
         for value in values:
-            masked: str = self.algorithm_function(value, salt)  # type: ignore
+            if value is None:
+                masked_values.append(None)
+                continue
+
+            masked: str = self.algorithm_function(str(value), salt)  # type: ignore
```

Comment:
Seeking opinions regarding how masking strategies should behave when used in DSR's.  The hash masking strategy is intended to work on strings, but when applied in a DSR context, there are many values that could be potentially passed here that are not strings.

Should we fail or turn them into strings?  Turning them into a string may cause the masking database query to fail if the type can't be coerced.  But is that the responsibility of the masking strategy or the responsibility of the connector/query config?

(As an aside, these bring up good use cases for our detailed Policy > Rule > Rule Target framework.  The customer could configure so string fields are targeted by the hash strategy and non string fields are targeted by null masking for example)

Reply:
very good questions, my opinion is that we should try to make it work, for now? the only benefit i'd see for failing explicitly is to get a better error to the user, so i wonder if we can _try to make it work_ in a way that we're still going to get as clear error messages as possible to the user if our attempt fails because of a restriction on the DB side (which is admittedly likely in many of the cases, i'm sure).

i agree with your point that this brings up the need to allow for more fine-grained rule management in our framework. that being said, the current capability to stratify masking rules pretty finely on data categories may not be too bad, in that certain data categories should _generally_ correspond to certain data types. but that's definitely not going to be flawless, so i agree that the framework could use some improvement to give the user more control in this area.

Reply:
Thanks so much for your comments @adamsachs !

I think we generally try to make the query - there's something here https://github.com/ethyca/fides/blob/main/src/fides/api/ops/service/connectors/query_config.py#L162 , but that only gets hit in limited cases if I recall.

But using this change as an example: if there's a date field in mongo that we were trying to mask with the hash masking strategy, now the hash strategy will convert the date to a string, and hash it and try to write it back.  With mongo, I think this request succeeds but that wouldn't hold for every db. 

More work to do in this area for sure: looking into what kinds of errors we surface to the user, expanding type support beyond basic field types, adding increased type coercion (in a DSR, say you need to use a field of one type to locate data of another type), and more fine-grained rule management as discussed.

Reply:
Totally agree with the decisioning here, thanks both!

One thing to add: in my opinion we'll need to allow "field-level" overrides as the escape hatch here. For example, if I know I have a field in my database that simply cannot be hashed, despite being a string, I could specify something inline like:
```
fides_meta:
  - data_type: string
    masking_strategy: null_rewrite
```

Reply:
agreed @NevilleS - i was actually considering linking to this (vague) issue we'd made a while back here https://github.com/ethyca/fides/issues/1499 - which i think gets at what you're saying.

(the refactoring of the policy/rule framework to allow for more control feels at least like an epic-level initiative, and this issue would be one piece of it. i think we still would need to sketch out a bunch of the other work, i don't know if we've got that documented anywhere!)

Reply:
@NevilleS Is this a feature that needs to added to this body of work?

Reply:
> Should a DSR fail if a hash masking strategy was specified to be used on a date field for example, or should we try to make it work?

It's a good question! In this example specifically I think it's a config. overhead, given otherwise we'd have to examine the meaning of hashing to say "if we set a timestamp to 00000... does that count as hashed?"

Reply:
Added Neville's comments to https://github.com/ethyca/fides/issues/1499 so field level overrides can be prioritized separately

Reply:
Yes, absolutely - my comment was more pontificating about the larger feature we could add to really give ourselves a lot more configuration flexibility, not something that needed to be added to this PR. Sorry!

Reply:
Oh no thanks for clarifying and for adding that suggestion! I think a field level override is going to be needed sooner rather than later